### PR TITLE
Update DeveleoperGuide and UserGuide to include maintainSorted

### PR DESCRIPTION
**Pull request by fustilio**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -572,6 +572,33 @@ Also creates coupling between the `Autocompleter` and `CommandBox`. +
...
+
+The list is sorted whenever a parcel is added, edited or if a redo command is made. This is
+because these commands are the ones that might possibly cause the new parcel to be placed
+in the wrong position. +
+
+To maintain the selected element despite sorting however, we require a way to track which parcel
+ws selected. Static variables to track which card has been selected has been added to the Model
+class. The Model class now has the following methods: `select`, `unselect`, `getPrevIndex`,
+`hasSelected` and `forceSelect`. Whenever we select an item, we will use select to indicate that
+we have selected a parcel card and we will then re-select that card after we have sorted. +
```
Comment (Kennard123661) [inline docs/DeveloperGuide.adoc]: Could you phrase this paragraph better? its quite ambiguous

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -572,6 +572,33 @@ Also creates coupling between the `Autocompleter` and `CommandBox`. +
 **Pros:** Easier to test since `CommandBoxTest` has already been set up and implemented. +
 **Cons:** `CommandBox` class now has multiple responsibilities, which violates SRP. +
 
+=== Maintain sorted
```
Comment (Kennard123661) [inline docs/DeveloperGuide.adoc]: Is there a better way you can name this? Like sorted mechanism (try to make it similar to the headings for the other features, which means ____ mechanism)

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -572,6 +572,33 @@ Also creates coupling between the `Autocompleter` and `CommandBox`. +
...
+**Aspect:** Implementation of `maintainSorted` +
+**Alternative 1 (current choice):** Constant sort the list of parcels whenever there is a change that
+potentially could disrupt the order of the list. +
+**Pros:** Intuitive and guarantees that list is sorted in the right order +
+**Cons:** Many commands have to be changed +
+**Alternative 2:** Insert the new / edited parcel to fit into the sorted list. +
+**Pros:** Use less computation as the list of parcels is already sorted. +
+**Cons:** More difficult to implement as we'll need to implement our own sorting algorithm as opposed to
+just using the built in sorting methods. +
+
```
Comment (Kennard123661) [inline docs/DeveloperGuide.adoc:765]: Are there more design considerations you could add?

**Review by Kennard123661 — CHANGES_REQUESTED** [review]
Please add a sequence diagram for each of the sorts for us to review @fustilio , please add your comments @CS2103AUG2017-T16-B1/developers

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (fustilio) [reply docs/DeveloperGuide.adoc:765]: Not really, but I'll try to think of more.

**Review by ktnc985 — APPROVED** [review]

**Pull request merged**